File: SWTOR HoloNet/SWTOR HoloNet/Module/Forum/Model/ForumPost.swift
//
//  ForumPost.swift
//  SWTOR HoloNet
//
//  Created by Ivan Fabijanovic on 16/10/14.
//  Copyright (c) 2014 Ivan Fabijanović. All rights reserved.
//

import UIKit

class ForumPost: Entity {
    
    // MARK: - Properties
    
    var avatarUrl: String?
    var username: String
    var date: String
    var postNumber: Int?
    var isBiowarePost: Bool
    var text: String
    var signature: String?
    
    // MARK: - Init
    
    init(id: Int, username: String, date: String, postNumber: Int?, isBiowarePost: Bool, text: String) {
        self.username = username
        self.date = date
        self.postNumber = postNumber
        self.isBiowarePost = isBiowarePost
        self.text = text
        super.init(id: id)
    }
   
}

File: android/app/src/main/java/com/ifabijanovic/holonet/forum/model/ForumPost.kt
package com.ifabijanovic.holonet.forum.model

import com.ifabijanovic.holonet.app.model.Entity

/**
 * Created by feb on 19/03/2017.
 */
data class ForumPost(
        override val id: Int,
        val avatarUrl: String?,
        val username: String,
        val date: String,
        val postNumber: Int?,
        val isBiowarePost: Boolean,
        val text: String,
        val signature: String?): Entity {
    constructor(
            id: Int,
            username: String,
            date: String,
            postNumber: Int?,
            isBiowarePost: Boolean,
            text: String): this(id, null, username, date, postNumber, isBiowarePost, text, null)
}
